File: create-icon.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw, ImageFont
import os

def create_icon():
    size = 32
    img = Image.new('RGBA', (size, size), (79, 70, 229, 255))
    draw = ImageDraw.Draw(img)

    # 绘制简单的图标
    # 背景已经是紫色
    # 添加白色文字效果
    font = None  # 使用默认字体

    # 在中间绘制一个白色方块作为"智"字的效果
    margin = 8
    draw.rectangle([margin, margin, size-margin, size-margin], fill=(255, 255, 255, 255))

    # 保存图标
    icon_path = os.path.join(os.path.dirname(__file__), 'public', 'icon.png')
    os.makedirs(os.path.dirname(icon_path), exist_ok=True)
    img.save(icon_path)
    print(f'✅ 图标已创建: {icon_path}')

    # 同时创建更大的图标
    large_size = 128
    large_img = Image.new('RGBA', (large_size, large_size), (79, 70, 229, 255))
    large_draw = ImageDraw.Draw(large_img)

    margin = 32
    large_draw.rectangle([margin, margin, large_size-margin, large_size-margin], fill=(255, 255, 255, 255))

    large_icon_path = os.path.join(os.path.dirname(__file__), 'public', 'icon.png')
    large_img.save(large_icon_path)
    print(f'✅ 大图标已保存: {large_icon_path}')

if __name__ == '__main__':
    try:
        create_icon()
        print('图标创建完成！')
    except Exception as e:
        print(f'❌ 创建图标失败: {e}')
        print('请安装 Python PIL 库: pip3 install Pillow')

File: create-icon.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// 创建简单的 PNG 图标
function createIcon() {
    const size = 32;
    const pixelCount = size * size * 4; // RGBA
    const buffer = Buffer.alloc(pixelCount);

    // 简单的紫色背景图标
    for (let y = 0; y < size; y++) {
        for (let x = 0; x < size; x++) {
            const offset = (y * size + x) * 4;

            // 紫色背景 (79, 70, 229 = #4F46E5)
            buffer[offset] = 79;     // R
            buffer[offset + 1] = 70;  // G
            buffer[offset + 2] = 229; // B
            buffer[offset + 3] = 255; // A

            // 简单的白色"智"字效果（中间白色区域）
            const centerX = size / 2;
            const centerY = size / 2;

            if (x > 8 && x < 24 && y > 8 && y < 24) {
                if ((x + y) % 2 === 0) {
                    buffer[offset] = 255;     // R
                    buffer[offset + 1] = 255;  // G
                    buffer[offset + 2] = 255;  // B
                    buffer[offset + 3] = 255;  // A
                }
            }
        }
    }

    // 写入 PNG 文件签名
    const pngSignature = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);
    const completePng = Buffer.concat([pngSignature, buffer]);

    const iconPath = path.join(__dirname, 'public', 'icon.png');
    fs.writeFileSync(iconPath, completePng);

    console.log('✅ 图标已创建:', iconPath);
}

// 创建更大的图标
function createLargeIcon() {
    const size = 128;
    const buffer = Buffer.alloc(size * size * 4);

    for (let y = 0; y < size; y++) {
        for (let x = 0; x < size; x++) {
            const offset = (y * size + x) * 4;

            // 紫色背景渐变
            const gradient = 1 - (Math.abs(x - size/2) + Math.abs(y - size/2)) / (size/2);
            const r = Math.floor(79 * gradient + 100);
            const g = Math.floor(70 * gradient + 100);
            const b = Math.floor(229 * gradient + 100);

            buffer[offset] = Math.min(255, Math.max(0, r));
            buffer[offset + 1] = Math.min(255, Math.max(0, g));
            buffer[offset + 2] = Math.min(255, Math.max(0, b));
            buffer[offset + 3] = 255;

            // 白色"智"字效果（中间白色区域）
            if (x > 32 && x < 96 && y > 32 && y < 96) {
                if ((x + y) % 3 === 0) {
                    buffer[offset] = 255;
                    buffer[offset + 1] = 255;
                    buffer[offset + 2] = 255;
                    buffer[offset + 3] = 255;
                }
            }
        }
    }

    const pngSignature = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);
    const completePng = Buffer.concat([pngSignature, buffer]);

    const iconPath = path.join(__dirname, 'public', 'icon.png');
    fs.writeFileSync(iconPath, completePng);

    console.log('✅ 大图标已创建:', iconPath);
}

try {
    createIcon();
    console.log('图标创建完成！');
} catch (error) {
    console.error('创建图标失败:', error);
    process.exit(1);
}
